tableauserverclient/server/endpoint/exceptions.py

```diff
@@ -23,40 +27,41 @@ def from_response(cls, resp, ns):
         return error_response
 
 
-class InternalServerError(Exception):
-    def __init__(self, server_response):
+class InternalServerError(TableauError):
+    def __init__(self, server_response, request_url):
         self.code = server_response.status_code
         self.content = server_response.content
+        self.url = request_url or "server"
 
     def __str__(self):
-        return "\n\nError status code: {0}\n{1}".format(self.code, self.content)
+        return "\n\nInternal error {0} at {1}\n{2}".format(self.code, self.url, self.content)
 
 
-class MissingRequiredFieldError(Exception):
+class MissingRequiredFieldError(TableauError):
```

Comment:
What's the purpose of this class (and the ones below) ?

Reply:
To be able to return more informative errors